python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -473,15 +475,25 @@ def agg(self, func):
 
         multilevel = _is_multi_agg(func)
         data = {}
-        for col_name, aggs, cols in zip(
-            column_names, included_aggregations, result_columns
+        for col_name, aggs, cols, orig_dtype in zip(
+            column_names,
+            included_aggregations,
+            result_columns,
+            orig_dtypes,
         ):
             for agg, col in zip(aggs, cols):
                 if multilevel:
                     agg_name = agg.__name__ if callable(agg) else agg
                     key = (col_name, agg_name)
                 else:
                     key = col_name
+                if (
+                    agg in {list, "collect"}
+                    and not is_categorical_dtype(orig_dtype)
```

Comment:
Why exclude `categorical_dtype`? It's not super obvious so we either should add a comment here, or possibly  `_with_type_metadata()` needs some additional logic?

Reply:
`cudf.ListDtype(categorical_dtype)` is `NotImplemented` (I think because the conversion from categorical dtypes to pyarrow types is not implemented). `ListDtype` stores the element type as a pyarrow type for which there isn't round-tripping for categorical dtypes right now.

Reply:
So I guess that right now I should probably disable list-based collection for categorical dtypes, since it doesn't really work:
```
In [1]: import cudf

In [2]: df = cudf.DataFrame({"a": cudf.Series(["a", "a", "b"]).astype("category"), "b": [1,2,3]})

In [3]: df
Out[3]: 
   a  b
0  a  1
1  a  2
2  b  3

In [4]: df.groupby("b").collect()
Out[4]: 
     a
b     
1  [0]
2  [0]
3  [1]

```

Reply:
Equally, pandas loses information in this scenario:
```
In [13]: df.to_pandas().groupby("b").agg(list)["a"].explode()
Out[13]: 
b
1    a
2    a
3    b
Name: a, dtype: object
```
Whereas:
```
In [16]: df.to_pandas().a
Out[16]: 
0    a
1    a
2    b
Name: a, dtype: category
Categories (2, object): ['a', 'b']
```

Reply:
Removed in d6c5d787aa